ServiceWebsite/ServiceWebsite.AcceptanceTests/Steps/RepresentativeQuestionnaireSteps.cs

```diff
@@ -104,44 +113,90 @@ public void InitiateJourneySteps(string page)
+            pageToValidate.Validate();
+            if(page == "about your computer" || page == "questionnaire completed")
             {
-                case "about your client":
-                    _aboutYourClient.Validate();
-                    break;
-                case "access to a suitable room":
-                    _accessToRoom.Validate();
-                    break;
-                case "client attendance":
-                    _clientAttendance.Validate();
-                    break;
-                case "hearing suitability":
-                    _hearingSuitability.Validate();
-                    break;
-                case "your computer":
-                    _yourComputer.Validate();
-                    break;
-                case "about your computer":
-                    _aboutYourComputer.Validate();
-                    _scenarioContext.Set<DecisionJourney>(_aboutYourComputer, "CurrentPage");
-                    break;
-                case "questionnaire completed":
-                    _questionnaireCompleted.Validate();
-                    _scenarioContext.Set<DecisionJourney>(_questionnaireCompleted, "CurrentPage");
-                    break;
-                case "please contact us":
-                    _pleaseContactUs.Validate();
-                    break;
-                case "thank you":
-                    _thankYou.Validate();
-                    break;
+                _scenarioContext.Set<DecisionJourney>((DecisionJourney)pageToValidate, "CurrentPage");
+            }
+        }
+
+        [Given(@"Representative participant starts the questionnaire")]
+        public void GivenRepresentativeParticipantStartsTheQuestionnaire()
+        {
+            _information.InformationScreen("Representative");
+        }
+
+        [When(@"provides answer")]
+        public void WhenProvidesAnswer(Table table)
+        {
+            var reponses = table.CreateSet<SuitabilityResponse>();
+            foreach (var response in reponses)
+            {
+               var currentPage = (DecisionJourney)GetPage(response.Page);
+                SelectAnswer(currentPage, response.Answer);
+                if(!string.IsNullOrEmpty(response.Details?.Trim()))
+                {
+                    currentPage.SelectYes(response.Details.Trim());
+                }
+                response.QuestionKey = currentPage.QuestionKey;
+                currentPage.Continue();
+            }
+            _scenarioContext.Set(reponses, "Responses");
+        }
+
+        [Then(@"all the answers should match")]
+        public void ThenAllTheAnswersShouldBeMatch()
+        {
+            var reponses = _scenarioContext.Get<IEnumerable<SuitabilityResponse>>("Responses");
+
+            foreach (var expectedResponse in reponses)
+            {
+                var displayedAnswer = GetMethods.GetText(By.XPath($"//td[@id='{expectedResponse.QuestionKey}']/strong"), _browserContext);
```

Comment:
if you can use css it's like a million times more performant than xpath. `By.Css("#${expectedResponse.QuestionKey} strong")`